stablehlo/transforms/ChloDecompositionPatternsMath.td

```diff
@@ -0,0 +1,335 @@
+                          StableHLO_ComparisonDirectionValue<"EQ">,
+                          (STABLEHLO_DEFAULT_COMPARISON_TYPE)))),
+                    (StableHLO_DivOp $x, $y),
+                    $zero),
+                  $xoy)))),
+          (StableHLO_SelectOp
+            (StableHLO_AndOp:$logical_and_lt_y_safe_min_lt_x_one
+              (StableHLO_CompareOp
+               $y,
+               (StableHLO_MulOp
+                 (StableHLO_SqrtOp
+                   (StableHLO_ConstantLikeSmallestNormalizedValue $signed_x)),
+                 (StableHLO_ConstantLike<"4"> $signed_x)),
+                StableHLO_ComparisonDirectionValue<"LT">,
+                (STABLEHLO_DEFAULT_COMPARISON_TYPE)),
+              (StableHLO_CompareOp
+               $x,
+               $one,
+                StableHLO_ComparisonDirectionValue<"LT">,
+                (STABLEHLO_DEFAULT_COMPARISON_TYPE))),
+            (StableHLO_DivOp
+              $y,
+              (StableHLO_SqrtOp:$sq
+                (StableHLO_MulOp
+                  (StableHLO_SelectOp:$am1
+                    $logical_and_lt_y_safe_min_lt_x_one,
+                    (StableHLO_NegOp
+                      (StableHLO_DivOp
+                        (StableHLO_MulOp $xp1, $xm1),
+                        (StableHLO_AddOp:$ap1 $a, $one))),
+                    (StableHLO_SelectOp:$x_ge_1_or_not
+                      (StableHLO_CompareOp
+                       $x,
+                       $one,
+                        StableHLO_ComparisonDirectionValue<"GE">,
+                        (STABLEHLO_DEFAULT_COMPARISON_TYPE)),
+                      (StableHLO_AddOp
+                        (StableHLO_DivOp:$divide_half_yy_rpxp1
+                          (StableHLO_MulOp:$half_yy $half, $yy),
+                          $rpxp1),
+                        (StableHLO_MulOp $half, $spxm1)),
+                      (StableHLO_SelectOp
+                        (StableHLO_CompareOp
+                         $a,
+                         (StableHLO_ConstantLike<"1.5"> $signed_x),
+                          StableHLO_ComparisonDirectionValue<"LE">,
+                          (STABLEHLO_DEFAULT_COMPARISON_TYPE)),
+                        (StableHLO_AddOp
+                          $divide_half_yy_rpxp1,
+                          (StableHLO_DivOp $half_yy, $smxm1)),
+                        (StableHLO_SubtractOp $a, $one)))),
+                  $ap1))),
+            (StableHLO_Log1pOp
+              (StableHLO_AddOp $am1, $sq))))),
+      $imag))>;
+
+// Arcus sine on real input:
+//
+//     arcsin(x) = 2 * arctan2(x, (1 + sqrt(1 - x * x)))
+//
+def : Pat<(CHLO_AsinOp NonComplexElementType:$x),
+  (StableHLO_MulOp
+    (StableHLO_ConstantLike<"2"> $x),
+    (StableHLO_Atan2Op
+      $x,
+      (StableHLO_AddOp
+        (StableHLO_ConstantLike<"1">:$constant_1 $x),
+        (StableHLO_SqrtOp
+          (StableHLO_SubtractOp
+            $constant_1,
+            (StableHLO_MulOp $x, $x))))))>;
```

Comment:
newline at EOF (likely to show up as lint error in CI)

Reply:
`./build_tools/github_actions/lint_whitespace_checks.sh -f`

Should fix this issue in all files in this PR

Reply:
Fixed here and other similar issues. Thanks for the lint_whitespace_checks.sh hint!